src/js/components/Anchor/Anchor.js

```diff
@@ -66,16 +66,33 @@ const Anchor = forwardRef(
     }, [children, icon, label]);
 
     let coloredIcon = icon;
-    if (icon && !icon.props.color) {
-      coloredIcon = cloneElement(icon, {
-        color: normalizeColor(
-          color ||
-            theme.anchor?.icon?.color ||
-            theme.anchor?.size?.[sizeProp || size]?.color ||
-            theme.anchor.color,
-          theme,
-        ),
-      });
+    if (icon) {
+      if (!icon.props.color) {
+        coloredIcon = cloneElement(icon, {
+          color: normalizeColor(
+            color ||
+              theme.anchor?.icon?.color ||
+              theme.anchor?.size?.[sizeProp || size]?.color ||
+              theme.anchor.color,
+            theme,
+          ),
+        });
+
+        if (!icon.props.size) {
+          const scaledSize = sizeProp || size || 'medium'; // fallback size
+
+          let iconSize;
+          if (sizeProp === 'xsmall') {
+            iconSize = 'small';
+          } else if (sizeProp === 'xxlarge') {
+            iconSize = 'xlarge';
+          } else {
+            iconSize = theme.anchor?.size?.[scaledSize]?.iconSize || scaledSize;
```

Comment:
I'm not seeing that we have `theme.anchor.size.[scaledSize].iconSize` as an available theme property

Reply:
she has `scaledSize` defined on line 82 we can change naming or leave 

       ` const scaledSize = sizeProp || size || 'medium'; // fallback size`

also not sure if we need the `medium` for fall back but we can discuss